### PR TITLE
Port nanite changes from /tg/

### PR DESCRIPTION
**Pull request by ghost**
:cl: Multiple /tg/ coders
add: Slimes and jellypeople pass nanites to their offspring. [Dingo-Dongler]
add: Added Nanite Sting nanite program, a manual trigger-based nanite spread. [Nebulacrity]
add: Added the Reduced Diagnostics nanite program: it disables the program list when scanned by portable nanite scanners, in return for a minor increase in replication speed. [XDTM]
tweak: The functionality has been removed from the Stealth program, but they can easily be stacked together if needed. [XDTM]
add: Added a new nanite program under Mesh Nanite Programming: Dermal Button. It gives an action button to hosts that signals their nanites when pressed. [XDTM]
balance: Viral Nanites now have a 100% chance to work, but only pulse every 7.5 seconds. [XDTM]
tweak: Stealth nanites grant immunity to viral nanites. [XDTM]
balance: Spreading nanites now have a higher chance to infect, but only pulse once every 5 seconds. [XDTM]
balance: Accelerated Regeneration now heals 0.5/s, (from 1) but only costs 0.5 nanites/s (from 2.5). [XDTM]
balance: Mechanical Repair now heals 1.5/s (from 1/s). [XDTM]
balance: Mind Control is now a Triggered program that costs 30, has a 180s cooldown, and lasts 60 seconds. [XDTM]
add: Mind control can now be triggered with a comm remote to input a custom objective. [XDTM]
add: Added the Happiness Enhancement and Happiness Suppression nanite programs, which boost or suppress the host's mood while active. [XDTM]
add: Sensor nanites (most of them) can now be installed as internal rules to programs, using their current setting. Rules must be met for the program to stay active, but it won't affect triggering. [XDTM]
del: The Nanite chamber can no longer install/uninstall programs, although it can still view them. [XDTM]
add: Instead, the nanite chamber can now easily destroy nanites from a host. [XDTM]
tweak: Viral nanites now disable cloud sync, if they don't modify Cloud ID, making them always effective. [XDTM]
balance: The Mitosis program now also works while connected to a cloud, although it will still generate errors in case the cloud sync stops working. [XDTM]
add: Added Species Sensor nanite program. [Nebulacrity]
add: Added the Distributed Computing nanite program, which slowly adds research points over time. [XDTM]
add: Added the Neural Network nanite program, which adds research points over time as well, but scales in efficiency for each mob that is currently running the program, breaking even with Distributed Computing at 12 hosts. [XDTM]
add: All of these programs only work on complex mobs (humans, monkeys, xenomorphs) and only have 25% efficiency on non-sentient mobs. [XDTM]
add: Nanites now have their own research point type, gained over time based on the amount of mobs with active nanites. [XDTM]
balance: Basic nanite programming is much cheaper and no longer requires basic robotics. [XDTM]
balance: All other nodes had their cost split into normal and nanite research, lowering their impact on the research progression for other departments. [XDTM]
add: Nanite public chambers can now swap Cloud IDs on existing nanites. [XDTM]
add: Nanite things now use tgui-next [actioninja]
/:cl:

### CODE REVIEW

**Review by yoyobatty — APPROVED** [review]
Doesn't look broken

**Comment by diegoxt45** [comment]
> add: Mind control can now be triggered with a comm remote to input a custom objective.

what

**Comment by yoyobatty** [comment]
Needs approval from designers/gms/has

**Comment by diegoxt45** [comment]
the mind control thing is only if you emag the program machine right?

**Comment by Manly-Dwarf** [comment]
tl;dr

**Comment by ghost** [comment]
@Manly-Dwarf nanites are cool again

also you can mine bitcoin- i mean research points with em. now science has a reason to spread them around.

**Comment by Toxici11i** [comment]
+1 please speedmerge

**Comment by karmaisblackandbluepilled** [comment]
+1 robot time
coom sock

**Comment by Manly-Dwarf** [comment]
+1 but you're responsible for all of the inevitable problems that arise
